### PR TITLE
[T4A7][T13-B4]Ng Huan Ran

### PR DESCRIPTION
**Pull request by shusiner**

### CODE REVIEW

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class NameTest {
+	
+	private Name testName;
+	private Name testName2;
```
Comment (z11i) [inline test/java/seedu/addressbook/data/person/NameTest.java:13]: You can have more descriptive name.

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	
+	@Before
+	public void setup() throws IllegalValueException {
+		testName = new Name("Obama");
+		testName2 = new Name("obama");
```
Comment (z11i) [inline test/java/seedu/addressbook/data/person/NameTest.java:18]: Good usage of `@Before`. You may want to rename your test class to a more specific one as other testing scenarios may not benefit from these class variables.

File: src/seedu/addressbook/data/person/Name.java [hunk 3]
```diff
@@ -43,6 +43,24 @@ public static boolean isValidName(String test) {
     public List<String> getWordsInName() {
         return Arrays.asList(fullName.split("\\s+"));
     }
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if: 
+     * -they are same disregarding if they are in capital letters(eg. John == john)
+     * -they are same up to reordering(eg. Tan Ah Kao == Ah Kao, Tan == Kao, Ah Tan and so on)
```
Comment (z11i) [inline src/seedu/addressbook/data/person/Name.java:50]: Is the reordering implemented?

**Comment by z11i** [comment]
Comments added. Nice work adding new features.

File: src/seedu/addressbook/data/person/Name.java (hunk 3 above)
Comment (shusiner) [reply src/seedu/addressbook/data/person/Name.java:50]: have not implemented it...

**Pull request closed without merging**